### PR TITLE
Migrate libnetwork to use netlink.Handle

### PR DESCRIPTION
**Pull request by aboch**
- ns pkg to provide a netlink handle over initNs
- osl sandbox to own its own netlink handle
- retrieve sandbox statistic via netlink

Signed-off-by: Alessandro Boch <aboch@docker.com>

### CODE REVIEW

**Comment by mrjana** [comment]
@aboch Can you please rebase?

File: ns/init_linux.go [hunk 1]
```diff
@@ -35,9 +45,21 @@ func SetNamespace() error {
 
 // ParseHandlerInt transforms the namespace handler into a integer
 func ParseHandlerInt() int {
-	return int(initNs)
+	return int(GetHandler())
+}
+
+// GetHandler returns the namespace handler
+func GetHandler() netns.NsHandle {
```
Comment (mrjana) [inline ns/init_linux.go]: Seems like this doesn't need to be exposed as this is only used internal to the package

File: osl/namespace_linux.go [hunk 2]
```diff
@@ -262,6 +288,7 @@ func (n *networkNamespace) InvokeFunc(f func()) error {
 func InitOSContext() func() {
 	runtime.LockOSThread()
 	nsOnce.Do(ns.Init)
+	runtime.LockOSThread()
```
Comment (mrjana) [inline osl/namespace_linux.go]: Why do you need this extra `LockOSThread` call?

File: osl/interface_linux.go [hunk 3]
```diff
@@ -233,17 +228,24 @@ func (n *networkNamespace) AddInterface(srcName, dstPrefix string, options ...If
 
 	path := n.path
 	isDefault := n.isDefault
+	nlh := n.nlHandle
+	nlhH := ns.NlHandle()
```
Comment (mrjana) [inline osl/interface_linux.go]: Can we just rename it `nlhHost`? `nlhH` seems confusing

Comment (aboch) [inline osl/interface_linux.go]: Will rename.

File: osl/namespace_linux.go (hunk 2 above)
Comment (aboch) [inline osl/namespace_linux.go]: Because it would get unlocked in ns.init during call to `netlink.NewHandle()`

File: ns/init_linux.go (hunk 1 above)
Comment (aboch) [inline ns/init_linux.go]: Right, I was using it in a previous diffs. Will make it unexported.

**Comment by aboch** [comment]
@mrjana Rebased and taken care of your comments. PTAL.

File: osl/namespace_linux.go (hunk 2 above)
Comment (mrjana) [inline osl/namespace_linux.go]: Then you don't need the first one.

Comment (aboch) [inline osl/namespace_linux.go]: yeah... let me remove it

**Comment by mavenugo** [comment]
@aboch PR requires rebase.

**Comment by mrjana** [comment]
LGTM

**Comment by mavenugo** [comment]
LGTM

**Pull request merged**